src/cli.rs

```diff
@@ -22,6 +22,9 @@ pub struct Cli {
     /// on a TTY as your non-main compositor instance, to avoid messing up the global environment.
     #[arg(long)]
     pub session: bool,
+    /// File Descriptor to which Niri sends its ready notification
+    #[arg(long)]
+    pub notify_fd: Option<i32>,
```

Comment:
> Pre-open a socket for the service and pass it to the service using the systemd activation protocol. **This by itself does not give so called "socket activation"**, but does allow any process trying to connect to the specified socket to do so immediately after the service is started (even before the service process is properly prepared to accept connections).

This allows processes that connect to the socket to get notified, instead of dinit itself notifying those processes.

Reply:
Did the link work right?

![image](https://github.com/YaLTeR/niri/assets/1794388/10e61d00-d0f7-44e7-b9ff-08758eb3a0dc)

Reply:
Oh, I thought you sent the wrong link by mistake. I thought you were referring to `socket_listen`.
Do you mean that I can set `ready-notification = pipevar:NOTIFY_SOCKET`? If so, then it won't work since NOTIFY_SOCKET will contain the FD of the pipe it created for ready notifications, not the path.

Reply:
Hm, so it kinda works similarly to systemd but not really? It might still be better to go through the env variable for some consistency with the systemd code?

Reply:
I think env variable is less error prone than a CLI argument in this case. So something like check if `NOTIFY_FD` env exists, if so write `READY=1` there and unset it, what do you think?

Reply:
> I think env variable is less error prone than a CLI argument in this case.
Could you elaborate on that? Env variables can be set by any process in the hierarchy, so I don't see how it can be less error prone.

Reply:
My thinking is:
1. The user can read about `--notify-fd` in the CLI help, try to set it, then weird things may happen. With env this is not that big of a problem because we can put the documentation on the wiki where less people who don't need to know about it will read about it.
2. As far as I understand the dinit help text above, for `pipevar` dinit will pick and set the FD by itself, whereas for `pipefd` you need to pick it and then make sure you pass the same number to niri?

Reply:
Oh, Alright, I can do that

Reply:
Updated